src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -544,6 +544,30 @@ void Simulator::poll_topics()
 	if (updated) {
 		orb_copy(ORB_ID(vehicle_status), _vehicle_status_sub, &_vehicle_status);
 	}
+
+	orb_check(_tune_control_sub, &updated);
+
+	if (updated) {
+
+		orb_copy(ORB_ID(tune_control), _tune_control_sub, &_tune_control);
+
+		_tunes.set_control(_tune_control);
+
+		mavlink_play_tune_t tune_msg = { };
+		mavlink_message_t message = { };
+
+		struct {
+			unsigned int frequency;
+			unsigned int duration;
+			unsigned int silence;
+		} tune_info = {};
+
+		_tunes.get_next_tune(tune_info.frequency, tune_info.duration, tune_info.silence);
```

Comment:
It's not enough to call this only when the topic gets updated. I think we need similar logic as in ToneAlarm.cpp.

Reply:
@bkueng got it, ill take a look at the logic and try to replicate it here.

Reply:
So for some reason I was thinking that I needed the tune control class to implement this.  As @bkueng pointed out it's not necessary, it's up to the receiving end/driver to buffer and queue the notes with the correct durations. (I already do this in the class I implemented for jMavSim)  Anyways, I have removed the tune control class.

Reply:
It should be fixed now, Thanks for the help.